nautobot_ssot_infoblox/diffsync/adapters/infoblox.py

```diff
@@ -24,21 +24,22 @@ class InfobloxAdapter(DiffSync):
 
     top_level = ["prefix", "ipaddress", "vlangroup", "vlan"]
 
-    def __init__(self, *args, job=None, sync=None, **kwargs):
+    def __init__(self, *args, job=None, sync=None, conn=None, **kwargs):
         """Initialize Infoblox.
 
         Args:
             job (object, optional): Infoblox job. Defaults to None.
             sync (object, optional): Infoblox DiffSync. Defaults to None.
+            conn (object, optional): InfobloxAPI connection. Defaults to None.
         """
         super().__init__(*args, **kwargs)
         self.job = job
         self.sync = sync
-        self.conn = InfobloxApi()
+        self.conn = conn
```

Comment:
I think it's better that we pass an initialized client into the adapter. This allows for multiple end-points to be used with the adapter. I've added some handling in case the connection is passed as None or is False.